treeherder/webapp/api/push.py

```diff
@@ -272,6 +272,12 @@ def health_summary(self, request, project):
             lint_failure_count = len(push_health_lint_failures)
             test_in_progress_count = None
 
+            status = push.get_status()
+            total_failures = test_failure_count + build_failure_count + lint_failure_count
+            # Override the testfailed value added in push.get_status so that it aligns with how we detect lint, build and test failures
+            # for the push health API's (total_failures doesn't include known intermittent failures)
+            status['testfailed'] = total_failures
```

Comment:
This is probably not something to address here or now, but this makes me wonder if the `Push` model should have a new status that takes into account test, build & lint, instead of re-using `testfailed`, especially since this is being done in more than one place. (I realize that this would probably be a fairly invasive change!)

Reply:
So do you mean showing each of those failures as separate build, lint and test failure categories in the pie chart (end consumer for this API)?

Reply:
No, I'm not suggesting any UI changes, just thinking that instead of reusing/abusing `testfailed` for counting total failures (I think right now it really is only _test_ failures?), a new field could be used instead to avoid breaking any current assumptions about it.

Reply:
Ah, I think I see what you're saying and that does sound like a reasonable change. It'd still override `testfailed` with the `test_failure_count` so that its using the same data source for those types of failures, but I'll add separate fields for `lintfailed` and `buildfailed` and tally them in the UI.

Reply:
Actually, I took another look at the code and even though its called "testfailed", the original value we're overriding does include lint and built failures. Its just being changed because of a discrepancy with the test failure count. Probably it should've been named`totalfailures` but... oh well.

Reply:
Hah, that's convenient!